src/lib/controllers/ui-controller.js

```diff
@@ -666,7 +666,13 @@ export default class UIController {
         },
 
         setWordLists (state, wordLists) {
-          let checkWordLists = Array.isArray(wordLists) ? wordLists : Object.values(wordLists)
+          let checkWordLists
+          if (!wordLists || (!Array.isArray(wordLists) && Object.keys(wordLists).length === 0)) {
+            checkWordLists = []
+          } else {
+            checkWordLists = Array.isArray(wordLists) ? wordLists : Object.values(wordLists)
+          }
+
```

Comment:
I don't really understand how this change fixes the problem.  If `setWordLists` is called with a single list that is populated, wouldn't checkWordLists get set to the wrong thing? I.e. the contents of the wordlist?

Reply:
The problem here is the following:
when the wordlist becomes empty - it passes 2 variants of result:
   - `null` (before this change it was `[null]`), if the list was cleared on by one 
   - `{}` (before this change it was `[{}]`), if the list was deleted completely by delete all

the second variant gave `state.hasWordListsData = true`, because the array was not empty really.
I fixed it with removing wrapping to array (to reduce convertions) and adding dditional check for empty object - that's why it sloved the problem.

Reply:
ok, I understand that, but wonder if this breaks the scenario where it gets passed a populated wordlist.  
In WordListController.onTextQuoteSelectorReceived we have this code:

```
      WordlistController.evt.WORDLIST_UPDATED.pub(this.getWordList(wordItem.languageCode))
```

Wouldn't this scenario cause checkWordLists here to be incorrectly set to the contents of the single wordlist?